curated_transformers/generation/logits.py

```diff
@@ -121,3 +122,43 @@ def _process_logits(self, logits: Tensor):
             return
 
         logits /= self.temperature
+
+
+class MaskTransform(LogitsTransform):
+    """
+    Set the probability of specific classes to zero.
+    """
+
+    def __init__(self, classes_to_mask: List[int]):
+        """
+        Construct a mask logits transform.
+
+        :param classes_to_mask:
+            Identifers pertaining to the classes that need to be masked.
+            An empty list results in a no-op.
+        """
+        super().__init__()
+        self.classes_to_mask = torch.tensor(
+            classes_to_mask, dtype=torch.long
+        )  # `torch.int32` isn't supported in indexing operations prior in torch<2.0.0.
+        if self.classes_to_mask.dim() != 1:
+            raise ValueError("Class selector tensor for mask transform must be 1D")
```

Comment:
I think in the case of generation the classes are always vocab items, so I think it's clearer if the errors mention that. Something like:

```suggestion
            raise ValueError("Vocabulary items mask must be 1D")
```

the same of all the mentions of _class_ below.

Reply:
Fixed.